casper/src/main/scala/coop/rchain/casper/engine/NodeRunning.scala

```diff
@@ -230,9 +230,15 @@ class NodeRunning[F[_]
   private def checkBlockReceived(hash: BlockHash): F[Boolean] =
     BlockStore[F].contains(hash)
 
+  /**
+    * Check if block is added to the DAG
+    */
+  private def checkBlockValidated(hash: BlockHash): F[Boolean] =
+    BlockDagStorage[F].getRepresentation.map(_.contains(hash))
+
   def handle(peer: PeerNode, msg: CasperMessage): F[Unit] = msg match {
     case h: BlockHashMessage =>
-      handleBlockHashMessage(peer, h)(checkBlockReceived)
+      handleBlockHashMessage(peer, h)(checkBlockValidated)
```

Comment:
If we only check if block is validated then block can be requested again from the network even if it's already stored in the block store which means it will be received and disregarded. 

If block is already in the block store it can be loaded here and immediately send to `incomingBlocksQueue`.